### PR TITLE
feat(label): added compact variant

### LINKED ISSUE #4160: Horizontal card grid demo updates
* The card title/toggle alignment changes between the collapsed/expanded state because the labels are added/removed between the states. Let's see if we can keep that alignment consistent between the two states.
Screen Shot 2021-06-17 at 11 16 11 AM
Screen Shot 2021-06-17 at 11 16 14 AM

### PR DESCRIPTION
**Pull request by mattnolting**
closes #4160

### CODE REVIEW

**Review by mcoker — CHANGES_REQUESTED** [review]
Looks good! I'm assuming you can have icons and an optional close button? If so, the current close button overflows the label. I suppose that's OK given the target area for the button would be tiny if it were contained inside the label. cc @mceledonia 

Screen Shot 2021-09-21 at 1 59 30 PM
Screen Shot 2021-09-21 at 1 59 11 PM

**Comment by mceledonia** [comment]
I think that's fine @mcoker I would rather have a larger click target than a smaller one for a11y purposes.

**Comment by mcarrano** [comment]
There is currently an open design issue for this here: https://github.com/patternfly/patternfly-design/issues/1048. @gonoraj can you take a look at this and make sure it aligns with what you were thinking?

File: src/patternfly/components/Label/label.scss [hunk 1]
```diff
@@ -112,6 +112,13 @@
   --pf-c-label--m-overflow__content--link--focus--before--BorderWidth: var(--pf-global--BorderWidth--md);
   --pf-c-label--m-overflow__content--link--focus--before--BorderColor: var(--pf-global--BorderColor--300);
 
+  // Compact
+  --pf-c-label--m-compact--PaddingTop: 0;
+  --pf-c-label--m-compact--PaddingRight: var(--pf-global--spacer--sm);
+  --pf-c-label--m-compact--PaddingBottom: 0;
+  --pf-c-label--m-compact--PaddingLeft: var(--pf-global--spacer--sm);
+  --pf-c-label--m-compact--FontSize: #{pf-font-prem(12px)};
```
[the remark this answers is not in the document]
Comment (mcoker) [reply src/patternfly/components/Label/label.scss]: @mattnolting ping - can you use the global var here?

**Review by mcoker — CHANGES_REQUESTED** [review]

**Review by mcarrano — APPROVED** [review]
Looks good to me! Thanks @mattnolting .

**Review by mcoker — APPROVED** [review]
🏆

**Pull request merged**